File: Kol_2/2024_Septembar/Server/Models/Racun.cs
using System.Text.Json.Serialization;

public class Racun
{
    private const int OsnovicaZaStruju = 150;
    private const int OsnovicaZaKomunalije = 100;

    [Key]
    public int ID { get; set; }

    public required Stan Stan { get; set; }

    [Range(1,12)]
    public int Mesec { get; set; }

    public bool Placen { get; set; }

    public int CenaVode { get; set; }

    public int CenaZajednickeStruje
    {
        get =>
            Stan.BrojClanova * OsnovicaZaStruju;
    }
    public int CenaKomunalija {
        get =>
            Stan.BrojClanova * OsnovicaZaKomunalije;
    }
}

File: Kol_2/2024_Septembar/Client/app/models/racun.js
export class Racun {
  constructor(mesec, cenavode, cenaZajednickeStruje, cenaKomunalija, placen) {
    this.mesec = mesec;
    this.cenavode = cenavode;
    this.cenaZajednickeStruje = cenaZajednickeStruje;
    this.cenaKomunalija = cenaKomunalija;
    this.placen = placen;
  }

  // ovo nije dobro sto se u modelu radi crtanje ali ok tad nismo znali bolje a i realno
  // oni na casovima i pripremama nisu nista bolji

  draw(container) {
    const box = document.createElement("div");
    box.classList.add("racunBox");
    if (this.placen == "Da") {
      box.classList.add("placen");
    } else {
      box.classList.add("neplacen");
    }

    const polja = [
      {
        label: "Mesec:",
        value: this.mesec,
      },
      {
        label: "Voda:",
        value: this.cenavode,
      },
      {
        label: "Zajednička struja:",
        value: this.cenaZajednickeStruje,
      },
      {
        label: "Komunalne usluge:",
        value: this.cenaKomunalija,
      },
      {
        label: "Plaćen:",
        value: this.placen,
      },
    ];

    polja.forEach((polje) => {
      let lbl = document.createElement("label");
      lbl.innerHTML = polje.label;

      let lblValue = document.createElement("label");
      lblValue.innerHTML = polje.value;

      box.append(lbl, lblValue);
    });

    container.appendChild(box);
  }
}
